.github/workflows/qawolf/getInbox.js

```diff
@@ -0,0 +1,134 @@
+const axios = require("axios");
+const retry = require("async-retry");
+const getUrls = require("get-urls");
+
+const emailQuery = `
+query email($created_after: String!, $to: String!) {
+  email(created_after: $created_after, to: $to) {
+    from
+    html
+    subject
+    text
+    to
+  }
+}
+`;
+
+const pollForEmail = async ({
+  apiKey,
+  createdAfter,
+  teamId,
+  timeoutMs,
+  to,
+}) => {
+  let timeout = false;
+
+  const requestPromise = retry(
+    async (_, attempt) => {
+      if (timeout) return;
+
+      try {
+        console.log(`wait for email to ${to}, attempt ${attempt}`);
+
+        const result = await axios.post(
+          "https://www.qawolf.com/api/graphql",
+          {
+            query: emailQuery,
+            variables: {
+              created_after: createdAfter.toISOString(),
+              to,
+            },
+          },
+          { headers: { authorization: apiKey, "x-qawolf-team-id": teamId } }
+        );
+
+        const { data, errors } = result?.data || {};
+        if (errors?.length > 0) {
+          throw new Error("GraphQL Errors " + JSON.stringify(errors));
+        }
+
+        const email = data?.email;
+        if (!email) throw new Error(`email to ${to} not received`);
+
+        console.log(`found email to ${to} with subject ${email.subject}`);
+
+        return email;
+      } catch (e) {
+        console.debug(
+          `pollForEmail failed ${e} ${JSON.stringify(e.response?.data)}`
+        );
+        throw e;
+      }
+    },
+    {
+      factor: 1,
+      maxTimeout: 3000,
+      minTimeout: 3000,
+      retries: Math.round(timeoutMs / 3000),
+    }
+  );
+
+  const timeoutPromise = new Promise((_, reject) => {
+    setTimeout(() => {
+      timeout = true;
+      reject(new Error(`Email to ${to} not found`));
+    }, timeoutMs);
+  });
+
+  return Promise.race([requestPromise, timeoutPromise]);
+};
+
+function getInbox(args = {}) {
+  const apiKey = process.env.QAWOLF_API_KEY;
```

Comment:
Where are these keys sourced? Do we need to add them as GitHub env variables for this repo?

Reply:
They are already in GitHub env variables, since those are both needed for the sync.